holoviews/plotting/bokeh/raster.py

```diff
@@ -291,11 +292,15 @@ def get_data(self, element, ranges, style):
                         yc.append(ys.mean())
                 else:
                     mask.append(False)
+            mask = np.array(mask)
 
-            data = {'xs': XS, 'ys': YS, z.name: zvals[np.array(mask)]}
+            data = {'xs': XS, 'ys': YS, z.name: zvals[mask]}
             if 'hover' in self.handles:
-                data[x] = np.array(xc)
-                data[y] = np.array(yc)
+                if not self.static_source:
+                    hover_data = self._collect_hover_data(
+                            element, mask, transpose=self.invert_axes)
```

Comment:
Hey thanks!  Yes, that is correct. To be honest I know very little about Bokeh (thanks to HoloViews I don't need to 😁), and have no idea why it works like that... all I know is that when I replicate what was done for 1D-coords, it fills the values in backwards. Something like this:
![bad_labels](https://user-images.githubusercontent.com/31031841/223503039-a45e3bde-1910-40bf-8bba-f50b9a146649.png)
I did my best to be explicit with what is going on, but it is very unintuitive. Maybe it would be better to do something like `hover_data = self._collect_hover_data(element, mask, irregular=True)` and then have the `_collect_hover_data` method handle the logic of whether to transpose?

Reply:
I like that idea!

Reply:
Done! Let me know what you think.

Reply:
Looks good. Will merge when tests pass.

Reply:
Awesome, thanks for the speedy review!